ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUpdaterThread.java

```diff
@@ -646,6 +668,10 @@ public boolean runOneWorkerIteration(
     return true;
   }
 
+  private synchronized boolean isDbPresentInFailoverSet(String dbName) {
+    return dbsBeingFailedOver.containsKey(dbName) && dbsBeingFailedOver.get(dbName);
```

Comment:
This will not give the desired protection. The atomicity part is still vulnerable